plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/TimestampDecoder.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.decoders;
+
+import io.trino.plugin.pinot.conversion.PinotTimestamps;
+import io.trino.spi.TrinoException;
+import io.trino.spi.block.BlockBuilder;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.function.Supplier;
+
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static java.lang.String.format;
+import static java.time.ZoneOffset.UTC;
+
+public class TimestampDecoder
+        implements Decoder
+{
+    @Override
+    public void decode(Supplier<Object> getter, BlockBuilder output)
+    {
+        Object value = getter.get();
+        if (value == null) {
+            output.appendNull();
+        }
+        else {
+            LocalDateTime timestamp;
+            if (value instanceof String) {
+                String valueString = (String) value;
+                timestamp = PinotTimestamps.tryParse(valueString);
+                if (timestamp == null) {
+                    throw new TrinoException(NOT_SUPPORTED, format(
+                            "Unable to parse string representation of type TIMESTAMP: %s [%s]",
+                            value,
+                            value.getClass().getSimpleName()));
+                }
+            }
+            else if (value instanceof Number) {
+                timestamp = LocalDateTime.ofInstant(Instant.ofEpochMilli(((Number) value).longValue()), UTC);
```

Comment:
Avoid casts from `Number`. `Byte`, `BigDecimal` etc. are all subclasses of `Number`.

Reply:
@hashhar @xiangfu0 Pinot's MAX function returns double https://docs.pinot.apache.org/users/user-guide-query/supported-aggregations I think the options are either keeping `Number` or having 2 conditionals: one for `Double` and one for `Long`. I rather keep checking for `Number` type, let me know your thoughts

Reply:
I think number is ok.

Reply:
having `else if (value instanceof Double || value instanceof Long)` is still useful in my opinion.

Reply:
this is a temporary conversion to micros to be able invoke `TIMESTAMP_MILLIS.writeLong(output, epochMicros);` if there is a better way to write milliseconds to the output block please let me know. Please read the javadocs in `ShortTimestampType`

Reply:
I've now made this change